lib/winston/logger.js

```diff
@@ -24,17 +24,32 @@ const config = require('./config');
  */
 class Logger extends Transform {
   /**
-   * Constructor function for the Logger object responsible for persisting log
-   * messages and metadata to one or more transports.
-   * @param {!Object} options - foo
-   */
+     * Constructor function for the Logger object responsible for persisting log
+     * messages and metadata to one or more transports.
+     * @param {!Object} options - foo
+     */
   constructor(options) {
     super({
       objectMode: true
     });
     this.configure(options);
   }
 
+  child(defaultRequestMetadata) {
+    const logger = this;
+    return Object.create(logger, {
+      write: {
+        value: function (info) {
+          logger.write(Object.assign(
```

Comment:
if info is a Error type value,this will miss some values,such as stack.

Reply:
@little-yao Thank you for noticing this! I've fixed the case when error is passed as the only param; it doesn't look like Winston itself supports cases when you pass error message as string followed by Error message, but please correct me if I'm wrong or if there are additional cases that need handling. Also would appreciate if you could check latest commit for correctness.

Reply:
Changed, thanks!